core/sdk.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2023 Mia srl
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rond-authz/rond/internal/metrics"
+	"github.com/rond-authz/rond/openapi"
+	"github.com/rond-authz/rond/types"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+	"github.com/uptrace/bunrouter"
+)
+
+type PolicyResult struct {
+	QueryToProxy []byte
+	Allowed      bool
+}
+
+// Warning: This interface is experimental, and it could change with breaking also in rond patches.
+// Does not use outside this repository until it is not ready.
+type SDK interface {
+	Metrics() metrics.Metrics
+
+	FindEvaluator(logger *logrus.Entry, method, path string) (SDKEvaluator, error)
```

Comment:
I'd define a logging interface compatible with *logrus.Entry only defining the methods we are using so that we can remove the coupling with logrus already.

Reply:
At the moment, it's too complex to address also this. At the end of the sdk, I don't know also if we should use loggers in extensive way. 
When we use sdk/lib, it's not so common to have library logging.

Reply:
I agree, if the premise is that logging will be removed I'm ok with keeping it as is; maybe flag it to be removed in the interface?

Reply:
I'm not sure for this, I think we should think about depending on how it's used at the moment.